pkg/ccl/backupccl/backup_test.go

```diff
@@ -5673,6 +5673,129 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
+		runner.Exec(t, "ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds=1")
+		runner.Exec(t, "DROP INDEX foo@baz")
+		time.Sleep(time.Second * 2)
+
+		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
+		require.Equal(t, []string{"/Table/57/{1-4}"}, actualResolvedSpans)
+		runner.Exec(t, "DROP DATABASE test;")
+		actualResolvedSpans = nil
+	})
+
+	t.Run("interleaved-spans", func(t *testing.T) {
+		runner.Exec(t, "CREATE DATABASE test; USE test;")
+		runner.Exec(t, "CREATE TABLE grandparent (a INT PRIMARY KEY, v BYTES, INDEX gpindex (v))")
+		runner.Exec(t, "CREATE TABLE parent (a INT, b INT, v BYTES, "+
+			"PRIMARY KEY(a, b)) INTERLEAVE IN PARENT grandparent(a)")
+		runner.Exec(t, "CREATE TABLE child (a INT, b INT, c INT, v BYTES, "+
+			"PRIMARY KEY(a, b, c), INDEX childindex(c)) INTERLEAVE IN PARENT parent(a, b)")
+
+		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
+		// /Table/59/{1-2} encompasses the pk of grandparent, and the interleaved
+		// tables parent and child.
+		// /Table/59/2 - /Table/59/3 is for the gpindex
+		// /Table/61/{2-3} is for the childindex
+		require.Equal(t, []string{"/Table/59/{1-3}", "/Table/61/{2-3}"}, actualResolvedSpans)
+		runner.Exec(t, "DROP DATABASE test;")
+		actualResolvedSpans = nil
+	})
+
+	t.Run("revs-span-merge", func(t *testing.T) {
+		runner.Exec(t, "CREATE DATABASE test; USE test;")
+		runner.Exec(t, "CREATE TABLE foo (k INT PRIMARY KEY, v BYTES, name STRING, "+
+			"INDEX baz(name), INDEX bar (v))")
+		runner.Exec(t, "INSERT INTO foo VALUES (1, NULL, 'testuser')")
+		runner.Exec(t, "ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds=60")
+		runner.Exec(t, "DROP INDEX foo@baz")
+
+		runner.Exec(t, `BACKUP DATABASE test TO 'nodelocal://0/fooz' WITH revision_history`)
+
+		// The BACKUP with revision history will pickup the dropped index baz as
+		// well because it existed in a non-drop state at some point in the interval
+		// covered by this BACKUP.
+		require.Equal(t, []string{"/Table/63/{1-4}"}, actualResolvedSpans)
+		actualResolvedSpans = nil
+		runner.Exec(t, "DROP TABLE foo")
+
+		runner.Exec(t, "CREATE TABLE foo2 (k INT PRIMARY KEY, v BYTES, name STRING, "+
+			"INDEX baz(name), INDEX bar (v))")
+		runner.Exec(t, "INSERT INTO foo2 VALUES (1, NULL, 'testuser')")
+		runner.Exec(t, "ALTER TABLE foo2 CONFIGURE ZONE USING gc.ttlseconds=60")
+		runner.Exec(t, "DROP INDEX foo2@baz")
+
+		runner.Exec(t, `BACKUP DATABASE test TO 'nodelocal://0/fooz' WITH revision_history`)
+		// We expect to see only the non-drop indexes of table foo in this
+		// incremental backup with revision history. We also expect to see both drop
+		// and non-drop indexes of table foo2 as all the indexes were live at some
+		// point in the interval covered by this BACKUP.
+		require.Equal(t, []string{"/Table/63/{1-2}", "/Table/63/{3-4}", "/Table/64/{1-4}"}, actualResolvedSpans)
+		actualResolvedSpans = nil
+	})
```

Comment:
These test cases look like they cover most of the cases that I think are interesting.

I think that also a test case with the last index is GC'ed and another where the last index is dropped would be interesting.

I think it would also be interested in seeing some test cases for `spansForAllTableIndexes` with 2 non-interleaved adjacent tables:
- Where the first table has their last indexed dropped
- Where the first table has their last index gc'ed

I think adding these tests at a more unit level with passing in a mock `checkKVForBounds`.